listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -493,6 +500,59 @@ def unhide_user_timeline_event(user_name):
     return jsonify({"status": "ok"})
 
 
+@user_timeline_event_api_bp.route('/user/<user_name>/timeline-event/create/recommend-personal', methods=['POST', 'OPTIONS'])
+@crossdomain
+@ratelimit()
+def create_personal_recommendation_event(user_name):
+    '''
+    Make the user recommend a recording to their followers.
+    The request should post the following data about the recording being
+    recommended, and also the list of followers getting recommended:
+
+    .. code-block:: json
+    {
+        "metadata": {
+            "artist_name": "<The name of the artist, required>",
+            "track_name": "<The name of the track, required>",
+            "recording_msid": "<The MessyBrainz ID of the recording, required>",
+            "release_name": "<The name of the release, optional>",
+            "recording_mbid": "<The MusicBrainz ID of the recording, optional>",
+            "followers": [<Row IDs (integer) of the follower, required>]
+            "blurb_content": "<String containing personalized recommendation>"
+        }
+    }
+
+    :statuscode 200: Successful query, recording has been recommended!
+    :statuscode 400: Bad request, check ``response['error']`` for more
+    details.
+    :statuscode 401: Unauthorized, you do not have permissions to recommend
+    personal recordings on the behalf of this user
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+
+    '''
+    user = validate_auth_header()
+
+    if user_name != user['musicbrainz_id']:
+        raise APIUnauthorized("You don't have permissions to post to this user's timeline.")
+
+    try:
+        data = ujson.loads(request.get_data())
+    except ValueError as e:
+        raise APIBadRequest(f"Invalid JSON: {str(e)}")
+
+    metadata = data['metadata']
+
```

Comment:
There should be a is_following check here for all users, no? I'd suggest to create a new function db_follow which receives an array and checks that the users follows all of the other users. Also, the metadata will need to convert usernames  of the person to whom the track is recommended into user ids. 

We want this so that external api users can easily use this api (they don't have access to listenbrainz user ids , at least not by our intention). Secondly, so that everything keeps working smoothly when a user renames themself.